lib/api/controllers/remoteActions/prepareDb.js

```diff
@@ -33,45 +27,31 @@ module.exports = function PrepareDb (kuzzle, request) {
     this.defaultRoleDefinition = this.params.roleWithoutAdmin;
 
     this.kuzzle.pluginsManager.trigger('log:info', '== Starting DB preparation...');
-    createInternalStructure.call(this)
-      .then(() => {
-        return readFile.call(this, 'mappings');
-      })
-      .then(() => {
-        return readFile.call(this, 'fixtures');
-      })
-      .then(() => {
-        return createIndexes.call(this);
-      })
-      .then(() => {
-        return importMapping.call(this);
-      })
-      .then(() => {
-        return importFixtures.call(this);
-      })
+
+    return createInternalStructure.call(this)
+      .then(() => readFile.call(this, 'mappings'))
+      .then(() => readFile.call(this, 'fixtures'))
+      .then(() => createIndexes.call(this))
+      .then(() => importMapping.call(this))
+      .then(() => importFixtures.call(this))
       .then(() => {
         this.kuzzle.pluginsManager.trigger('log:info', '== DB preparation done.');
-        return deferred.resolve(new ResponseObject(request));
+        return {databasePrepared: true};
       })
       .catch(error => {
         this.kuzzle.pluginsManager.trigger('log:error', '!! An error occured during the process.\nHere is the original error object:\n', error);
-        return deferred.resolve(new ResponseObject(request, new InternalError(error)));
+        return q.reject(new InternalError(error));
       });
-  } else {
-    response = new ResponseObject(request);
-    response.data.body = { isWorker: true };
-    deferred.resolve(response);
-    return deferred.promise;
   }
 
-  return deferred.promise;
+  return q({isWorker: true});
```

Comment:
Why not?
I wanted to return relevant data and found nothing more interesting than that. :-)